subprojects/kotlin-dsl-plugins/src/test/kotlin/org/gradle/kotlin/dsl/plugins/precompiled/tasks/ExtractPrecompiledScriptPluginPluginsTest.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.plugins.precompiled.tasks
+
+import org.gradle.kotlin.dsl.fixtures.TestWithTempFiles
+import org.gradle.kotlin.dsl.fixtures.equalToMultiLineString
+
+import org.gradle.kotlin.dsl.plugins.precompiled.PrecompiledScriptPlugin
+
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.TypeSafeMatcher
+
+import org.junit.Test
+
+import java.io.File
+
+
+class ExtractPrecompiledScriptPluginPluginsTest : TestWithTempFiles() {
+
+    private
+    val outputDir by lazy {
+        newFolder("plugins")
+    }
+
+    @Test
+    fun `can extract plugins block from script with only plugins`() {
+
+        extractPluginsFrom(
+            scriptPlugin(
+                "plugins-only.gradle.kts",
+                """
+                // this comment will be removed
+                plugins {
+                    java
+                }
+                // and so will the rest of the script
+                """
+            )
+        )
+
+        assertThat(
+            outputFile("plugins-only.gradle.kts").readText(),
+            equalToMultiLineString("""
+                ${"// this comment will be removed".replacedBySpaces()}
+                plugins {
+                    java
+                }"""
+            )
+        )
+    }
+
+    @Test
+    fun `can extract plugins block from script with a buildscript block`() {
+
+        extractPluginsFrom(
+            // the `buildscript` block is not really valid in precompiled script plugins (causes a runtime error)
```

Comment:
❓ any chance we could remove it from the script template in order for the IDE/compiler to clearly early shows that it isn't supported?

Reply:
Unfortunately not. The current IDE hook for selection of script templates doesn't let us provide different templates based on the script location or role.

Reply:
Right, that's good enough for now